third_party/tlx/language/barrier.py

```diff
@@ -8,12 +8,30 @@ def alloc_barriers(
     num_barriers: tl.constexpr,
     arrive_count: tl.constexpr,
     _builder=None,
-) -> tlx.mbarriers:
+) -> tlx.buffered_tensor:
     """
     Allocates buffer in shared memory and initialize mbarriers with arrive_counts.
-    
-    Input: 
+
+    Input:
     - `num_barriers`: The number of barriers to allocate.
     - `arrive_counts`: The number of threads that need to arrive at the barrier before it can be released.
     """
-    return tlx.mbarriers(_builder.create_alloc_barriers(num_barriers.value, arrive_count.value), )
+    return tlx.buffered_tensor(_builder.create_alloc_barriers(num_barriers.value, arrive_count.value), )
+
+
+@tl.builtin
+def barrier_expect(
```

Comment:
Name it `barrier_expect_size` or something? There is another use of barrier expect which sets the expected number of threads that arrive at the barrier point.